Exception on application exit if MIDI device was disconnected

If a Roland FP-90 digital piano is turned off and the application is exited the following exception occurs:

```
Exception in thread "Thread-1" java.lang.IllegalStateException: Can't call send() with a closed receiver
	at uk.co.xfactorylibrarians.coremidi4j.CoreMidiReceiver.send(CoreMidiReceiver.java:57)
	at org.calinburloiu.music.microtuner.MidiTuner.tune(Tuner.scala:47)
	at org.calinburloiu.music.microtuner.MicrotonalistApp$$anon$1.org$calinburloiu$music$microtuner$LoggerTuner$$super$tune(MicrotonalistApp.scala:90)
	at org.calinburloiu.music.microtuner.LoggerTuner.tune(Tuner.scala:36)
	at org.calinburloiu.music.microtuner.LoggerTuner.tune$(Tuner.scala:33)
	at org.calinburloiu.music.microtuner.MicrotonalistApp$$anon$1.tune(MicrotonalistApp.scala:90)
	at org.calinburloiu.music.microtuner.MicrotonalistApp$$anon$2.run(MicrotonalistApp.scala:111)
```

Similar issue when switching tuning after disconnecting the device:

```
Exception in thread "AWT-EventQueue-0" org.calinburloiu.music.microtuner.MidiTunerException: Failed to send tune message to device! Did you disconnect the device?
	at org.calinburloiu.music.microtuner.MidiTuner.tune(Tuner.scala:51)
	at org.calinburloiu.music.microtuner.MicrotonalistApp$$anon$1.org$calinburloiu$music$microtuner$LoggerTuner$$super$tune(MicrotonalistApp.scala:90)
	at org.calinburloiu.music.microtuner.LoggerTuner.tune(Tuner.scala:36)
	at org.calinburloiu.music.microtuner.LoggerTuner.tune$(Tuner.scala:33)
	at org.calinburloiu.music.microtuner.MicrotonalistApp$$anon$1.tune(MicrotonalistApp.scala:90)
	at org.calinburloiu.music.microtuner.TuningSwitch.apply(TuningSwitch.scala:39)
	at org.calinburloiu.music.microtuner.TuningListFrame.$anonfun$new$1(TuningListFrame.scala:46)
	at javax.swing.JList.fireSelectionValueChanged(JList.java:1802)
	at javax.swing.JList$ListSelectionHandler.valueChanged(JList.java:1816)
	at javax.swing.DefaultListSelectionModel.fireValueChanged(DefaultListSelectionModel.java:184)
	at javax.swing.DefaultListSelectionModel.fireValueChanged(DefaultListSelectionModel.java:164)
	at javax.swing.DefaultListSelectionModel.fireValueChanged(DefaultListSelectionModel.java:211)
	at javax.swing.DefaultListSelectionModel.changeSelection(DefaultListSelectionModel.java:405)
	at javax.swing.DefaultListSelectionModel.changeSelection(DefaultListSelectionModel.java:415)
	at javax.swing.DefaultListSelectionModel.setSelectionInterval(DefaultListSelectionModel.java:459)
	at javax.swing.JList.setSelectedIndex(JList.java:2216)
	at javax.swing.plaf.basic.BasicListUI$Actions.changeSelection(BasicListUI.java:2182)
	at javax.swing.plaf.basic.BasicListUI$Actions.actionPerformed(BasicListUI.java:1866)
	at javax.swing.SwingUtilities.notifyAction(SwingUtilities.java:1668)
	at javax.swing.JComponent.processKeyBinding(JComponent.java:2882)
	at javax.swing.JComponent.processKeyBindings(JComponent.java:2929)
	at javax.swing.JComponent.processKeyEvent(JComponent.java:2845)
	at java.awt.Component.processEvent(Component.java:6316)
	at java.awt.Container.processEvent(Container.java:2239)
	at java.awt.Component.dispatchEventImpl(Component.java:4889)
	at java.awt.Container.dispatchEventImpl(Container.java:2297)
	at java.awt.Component.dispatchEvent(Component.java:4711)
	at java.awt.KeyboardFocusManager.redispatchEvent(KeyboardFocusManager.java:1954)
	at java.awt.DefaultKeyboardFocusManager.dispatchKeyEvent(DefaultKeyboardFocusManager.java:835)
	at java.awt.DefaultKeyboardFocusManager.preDispatchKeyEvent(DefaultKeyboardFocusManager.java:1103)
	at java.awt.DefaultKeyboardFocusManager.typeAheadAssertions(DefaultKeyboardFocusManager.java:974)
	at java.awt.DefaultKeyboardFocusManager.dispatchEvent(DefaultKeyboardFocusManager.java:800)
	at java.awt.Component.dispatchEventImpl(Component.java:4760)
	at java.awt.Container.dispatchEventImpl(Container.java:2297)
	at java.awt.Window.dispatchEventImpl(Window.java:2746)
	at java.awt.Component.dispatchEvent(Component.java:4711)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:760)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:84)
	at java.awt.EventQueue$4.run(EventQueue.java:733)
	at java.awt.EventQueue$4.run(EventQueue.java:731)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:730)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:205)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
Caused by: java.lang.IllegalStateException: Can't call send() with a closed receiver
	at uk.co.xfactorylibrarians.coremidi4j.CoreMidiReceiver.send(CoreMidiReceiver.java:57)
	at org.calinburloiu.music.microtuner.MidiTuner.tune(Tuner.scala:49)
	... 53 more
```

--- a/src/main/scala/org/calinburloiu/music/microtuner/MicrotonalistApp.scala
+++ b/src/main/scala/org/calinburloiu/music/microtuner/MicrotonalistApp.scala
@@ -108,7 +108,11 @@ object MicrotonalistApp extends StrictLogging {
     Runtime.getRuntime.addShutdownHook(new Thread() {
       override def run(): Unit = {
         logger.info("Switching back to 12-EDO before exit...")
-        tuner.tune(Tuning.Edo12)
+        try {
+          tuner.tune(Tuning.Edo12)
+        } catch {
+          case e: MidiTunerException => logger.error(e.getMessage)
+        }
         Thread.sleep(1000)
 
         midiManager.close()

--- a/src/main/scala/org/calinburloiu/music/microtuner/Tuner.scala
+++ b/src/main/scala/org/calinburloiu/music/microtuner/Tuner.scala
@@ -42,9 +42,14 @@ class MidiTuner(val receiver: Receiver,
 
   private val tuningMessageGenerator = tuningFormat.messageGenerator
 
+  @throws[MidiTunerException]
   override def tune(tuning: Tuning, baseNote: Int = 0): Unit = {
     val sysexMessage = tuningMessageGenerator.generate(tuning)
-    receiver.send(sysexMessage, -1)
+    try {
+      receiver.send(sysexMessage, -1)
+    } catch {
+      case e: IllegalStateException => throw new MidiTunerException(e)
+    }
   }
 
   // TODO Rethink which code component has the transpose responsibility
@@ -57,3 +62,6 @@ class MidiTuner(val receiver: Receiver,
       .toArray
   }
 }
+
+class MidiTunerException(cause: Throwable) extends RuntimeException(
+  "Failed to send tune message to device! Did you disconnect the device?", cause)

--- a/src/main/scala/org/calinburloiu/music/microtuner/TuningSwitch.scala
+++ b/src/main/scala/org/calinburloiu/music/microtuner/TuningSwitch.scala
@@ -18,16 +18,18 @@ package org.calinburloiu.music.microtuner
 
 import com.google.common.eventbus.EventBus
 import com.google.common.math.IntMath
+import com.typesafe.scalalogging.LazyLogging
+import org.calinburloiu.music.microtuner.MicrotonalistApp.logger
 import org.calinburloiu.music.tuning.{Tuning, TuningList}
 
 class TuningSwitch(val tuner: Tuner,
                    val tuningList: TuningList,
                    eventBus: EventBus,
-                   initialPosition: Int = 0) {
+                   initialPosition: Int = 0) extends LazyLogging {
 
   private[this] var _tuningIndex: Int = initialPosition
 
-  tuner.tune(currentTuning)
+  tune()
 
   def apply(index: Int): Unit = {
     if (_tuningIndex > tuningList.tunings.size - 1) {
@@ -36,7 +38,7 @@ class TuningSwitch(val tuner: Tuner,
       val oldTuningIndex = _tuningIndex
       _tuningIndex = index
 
-      tuner.tune(currentTuning)
+      tune()
       eventBus.post(TuningChangedEvent(tuningIndex, oldTuningIndex))
     }
   }
@@ -58,4 +60,12 @@ class TuningSwitch(val tuner: Tuner,
   def currentTuning: Tuning = tuningList(_tuningIndex)
 
   def tuningCount: Int = tuningList.tunings.size
+
+  protected def tune(): Unit = {
+    try {
+      tuner.tune(currentTuning)
+    } catch {
+      case e: MidiTunerException => logger.error("Failed to switch tuning: " + e.getMessage)
+    }
+  }
 }

